executor/include/execution/executor/inline_executor.hpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by Shrijit Singh on 2020-06-14.
+//
+
+#pragma once
+
+template <typename Interface, typename Cardinality, typename Blocking,
+          typename ProtoAllocator>
+struct inline_executor;
+
+template <>
+struct execution::is_executor_available<inline_executor> : std::true_type {};
+
+template <typename Interface, typename Cardinality, typename Blocking,
+          typename ProtoAllocator = std::allocator<void>>
+struct inline_executor
+    : public executor<inline_executor, Interface, Cardinality, Blocking,
+                      ProtoAllocator> {
+  using shape_type = std::size_t;
+
+  template <typename F, typename... Args>
+  void execute(F &&f, Args &&... args) {
+    std::invoke(std::forward<F>(f), std::forward<Args>(args)...);
```

Comment:
are executors exempt from having a return type? Or is this just deferred to be done later?

Reply:
one way executors have void return type

Reply:
Is there a case where executor have a proper return type which is not a future?

Reply:
Haven't seen that case before. It might exist, but if so, I'm unaware of it

Reply:
Yeah, I am looking into this currently, the two-way executor has been split off into another proposal.